atomic_reactor/plugins/pre_resolve_composes.py

```diff
@@ -0,0 +1,324 @@
+        self.compose_config.set_signing_intent(signing_intent['name'])
+
+    def forward_composes(self):
+        yum_repourls = [compose_info['result_repofile'] for compose_info in self.composes_info]
+        override_build_kwarg(self.workflow, 'yum_repourls', yum_repourls)
+
+    def make_result(self):
+        result = {
+            'composes': self.composes_info,
+            'signing_intent': self.compose_config.signing_intent['name'],
+            'signing_intent_overridden': self.compose_config.has_signing_intent_changed(),
+        }
+        self.log.debug('plugin result: %s', result)
+        return result
+
+    @property
+    def odcs_client(self):
+        if not self._odcs_client:
+            if self.odcs_openidc_secret_path:
+                token_path = os.path.join(self.odcs_openidc_secret_path, 'token')
+                with open(token_path, "r") as f:
+                    odcs_token = f.read().strip()
+            else:
+                odcs_token = None
+
+            self._odcs_client = ODCSClient(self.odcs_url, self.odcs_insecure, odcs_token)
+
+        return self._odcs_client
+
+    @property
+    def koji_session(self):
+        if not self._koji_session:
+            koji_auth_info = None
+            if self.koji_ssl_certs_dir:
+                koji_auth_info = {
+                    'ssl_certs_dir': self.koji_ssl_certs_dir,
+                }
+            self._koji_session = create_koji_session(self.koji_hub, koji_auth_info)
+
+        return self._koji_session
+
+
+class ComposeConfig(object):
+
+    def __init__(self, data, odcs_config, koji_tag=None):
+        data = data or {}
+        self.packages = data.get('packages', [])
+        self.koji_tag = koji_tag
+        self.odcs_config = odcs_config
+
+        signing_intent_name = data.get('signing_intent', self.odcs_config.default_signing_intent)
+        self.set_signing_intent(signing_intent_name)
+        self._original_signing_intent_name = signing_intent_name
+
+    def set_signing_intent(self, name):
+        self.signing_intent = self.odcs_config.get_signing_intent_by_name(name)
+
+    def has_signing_intent_changed(self):
+        return self.signing_intent['name'] != self._original_signing_intent_name
+
+    def render_request(self):
+        self.validate_for_request()
+
+        # For now, only 'tag' source type is supported.
+        request = {
+            'source_type': 'tag',
+            'source': self.koji_tag,
+            'packages': self.packages,
+            'sigkeys': self.signing_intent['keys']
+        }
+
+        return request
+
+    def validate_for_request(self):
+        """Verify enough information is available for requesting compose."""
+        if not self.packages:
+            raise ValueError('List of packages cannot be empty.')
+
+        if self.packages and not self.koji_tag:
+            raise ValueError('koji_tag is required when packages are used')
+
+
+class AbortResolveComposesPlugin(Exception):
```

Comment:
Is this only for cases where we want to skip the plugin? The name makes it sound like an error, but it isn't handled like an error.

Reply:
Yeah, it's meant to indicate cases where there isn't sufficient information for running the plugin (for instance, reactor-config doesn't include required info).
How about `SkipResolveComposesPlugin` ?

Reply:
I like that much better.

Reply:
Done.